front50-gcs/src/main/java/com/netflix/spinnaker/front50/config/GcsConfig.java

```diff
@@ -0,0 +1,174 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.front50.config;
+
+import com.netflix.spinnaker.front50.model.*;
+import com.google.api.services.storage.Storage;
+import com.google.api.services.storage.StorageScopes;
+import com.google.api.services.storage.model.Bucket;
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.HttpResponseException;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.client.RestTemplate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import rx.schedulers.Schedulers;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.concurrent.Executors;
+import java.util.Collections;
+
+@Configuration
+public class GcsConfig {
+  // Refresh every 10 minutes. In practice this either doesnt matter because refreshes are fast enough,
+  // or should be finer tuned. But it seems silly to refresh at a fast rate when changes are generally infrequent.
+  // Actual queries always check to see if the cache is out of date anyway. So this is mostly for the benefit of
+  // keeping other replicas up to date so that last-minute updates have fewer changes in them.
+  private static int APPLICATION_REFRESH_INTERVAL = 10 * 60;
+  private static int PROJECT_REFRESH_INTERVAL = 10 * 60;
+  private static int NOTIFICATION_REFRESH_INTERVAL = 10 * 60;
+  private static int PIPELINE_REFRESH_INTERVAL = 10 * 60;
+  private static int PIPELINE_STRATEGY_REFRESH_INTERVAL = 10 * 60;
+
+  @Value("${spinnaker.gcs.bucket}")
+  private String bucket;
+
+  @Value("${spinnaker.gcs.rootFolder}")
+  private String rootFolder;
+
+  @Value("${providers.google.primaryCredentials.jsonPath}")
+  private String jsonPath;
+
+  @Value("${providers.google.primaryCredentials.project}")
+  private String project;
+
+  @Value("${Implementation-Version:Unknown}")
+  private String applicationVersion;
```

Comment:
I'm not exactly sure where this is set, but it appears to be a standard Java convention if not provided by JDK itself. This is used to set the version in the UserAgent so that it is consistent with the debian package version as well.